### ISSUE TITLE
cmd/gofmt: extra indentations when adding comments  

### ISSUE THREAD

**Issue by a8m** [golang/go#32480]
I encounter an issue with `gofmt`, where adding a comment to the first line causes extra indentation to the rest of the lines. 

For example, giving this block of code:
```go
func main() {
	t1 := foo.Bar().
		Baz().
		Qux()
}
```
Adding a comment (`//`) after `Bar()` adds extra indentations to the lines below:
```go
func main() {
	t1 := foo.Bar(). //
				Baz().
				Qux()
}
```
[Playground](https://play.golang.org/p/defYUt_Zgmi)

Same for this case:
```go
func main() {
	t1 := foo.Bar(). // bar
				Baz(). // baz
				Qux()  // qux
}
```
[Playground](https://play.golang.org/p/PokqI18xpB3)

Maybe I miss something, but I expect the formatting to be as follow:
```go
func main() {
	t1 := foo.Bar(). // 
		Baz().
		Qux()
}

func main() {
	t1 := foo.Bar(). // bar
		Baz().   // baz
		Qux()    // qux
}
```

**Comment by agnivade** [comment]
Yes, adding a line comment after breaking the first line adds an extra tab. But I am curious whether you discovered this in some real-world situation or in a different case ?

Note that `gofmt` sometimes breaks if the code is written in a weird way. Leaving for @griesemer to take a call.

For reference, this happens with 1.12.4 and also tip at `go version devel +98100c56da Mon Jun 3 01:37:58 2019 +0000 linux/amd64`.

**Comment by a8m** [comment]
> But I am curious whether you discovered this in some real-world situation or in a different case ?

Yes, this example just have dummy methods for simplicity. But in my real-world example, I have longer names for these methods that get parameters, and adding line breaks make the code more clear to me.
Sometimes, it's also useful for documentation. For example, see the elasticsearch example in its [docs](https://godoc.org/github.com/olivere/elastic#example-Aggregations).

> Note that gofmt sometimes breaks if the code is written in a weird way.

I'm not sure what are you referring a weird code. The comments or the line breaks?

My current "hack" to solve this is to change the code block below:  
```go
func main() {
	result, err := client.Search(). // create a search client.
					Index("twitter").                  // search in index "twitter"
					Query(elastic.NewMatchAllQuery()). // return all results, but ...
					SearchType("count").               // ... do not return hits, just the count
					Aggregation("timeline", timeline). // add our aggregation to the query
					Pretty(true).                      // pretty print request and response JSON
					Do(context.Background())           // execute
}
```
To this (adding line break between `client.` and `Search()`:
```go
func main() {
	result, err := client.
		Search().                          // create a search client.
		Index("twitter").                  // search in index "twitter"
		Query(elastic.NewMatchAllQuery()). // return all results, but ...
		SearchType("count").               // ... do not return hits, just the count
		Aggregation("timeline", timeline). // add our aggregation to the query
		Pretty(true).                      // pretty print request and response JSON
		Do(context.Background())           // execute

}
```
[Playground](https://play.golang.org/p/D0f-k10eHuw)

I would be happy to fix/change this, but let's wait for @griesemer input before.

**Comment by agnivade** [comment]
Indeed, ES API is written this way. I was referring to the line breaks.

**Comment by griesemer** [comment]
Looks like something we should try to fix.

**Comment by a8m** [comment]
@griesemer, would you mind if I take it?

**Comment by griesemer** [comment]
@a8m You're welcome to give it a shot. But to be fair, I should let you know that I won't have time to help with this or answer detailed questions through the end of July at least. Also any review will have to wait at least that long.
